src/DataDictionary/search/DictionarySearcher/DictionarySearcher.jsx

```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AutoComplete from '@gen3/ui-component/dist/components/AutoComplete';
+import { prepareSearchData, searchKeyword, getSearchSummary } from './searchHelper';
+import './DictionarySearcher.css';
+
+class DictionarySearcher extends React.Component {
+  constructor(props) {
+    super(props);
+    this.searchData = prepareSearchData(props.dictionary);
+    this.autoCompleteRef = React.createRef();
+    this.state = {
+      suggestionList: [],
+      isSearchFinished: false,
+      searchResult: {
+        matchedNodes: [],
+        summary: {},
+      },
+      searchError: '',
+      hasError: false,
+    };
+    if (this.props.currentSearchKeyword) {
+      this.search(this.props.currentSearchKeyword);
```

Comment:
why is there a function being called in the constructor? I think this should be called elsewhere